packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -448,6 +448,10 @@ class AnimationController extends Animation<double>
   /// controller's ticker might get muted, in which case the animation
   /// controller's callbacks will no longer fire even though time is continuing
   /// to pass. See [Ticker.muted] and [TickerMode].
+  ///
+  /// If the animation was stopped (e.g. with [stop] or by setting a new [value]),
+  /// [isAnimating] will return `false` but the [status] will not change,
+  /// so the value of [AnimationStatus.isAnimating] might still be `true`.
```

Comment:
I probably miss something, it feels weird that one can't rely on this getter?
As it seemed to work before https://github.com/flutter/flutter/pull/148570 maybe there is a more generic way to fix it?

Reply:
The code in drawer.dart didn't rely on this getter before #148570; both this PR and the original code used the `_controller.status`.

At the time, `AnimationStatus` didn't have any "enhanced enum" features, so it just used a switch statement:

```dart
    if (widget.isDrawerOpen != oldWidget.isDrawerOpen) {
      switch (_controller.status) {
        case AnimationStatus.completed:
        case AnimationStatus.dismissed:
          _controller.value = widget.isDrawerOpen ? 1.0 : 0.0;
        case AnimationStatus.forward:
        case AnimationStatus.reverse:
          break;
      }
    }
```